seals/src/txout/blind.rs

```diff
@@ -131,115 +123,84 @@ impl<Id: SealTxid, M: SealCloseMethod> TxoSeal<M> for BlindSeal<Id, M> {
     }
 }
 
-impl<Id: SealTxid> BlindSeal<Id, CloseMethod> {
-    /// Creates new seal using TapretFirst closing method for the provided
-    /// outpoint. Uses `thread_rng` to initialize blinding factor.
-    pub fn tapret_first_rand_from(outpoint: Outpoint) -> Self {
-        BlindSeal::tapret_first_rand(outpoint.txid, outpoint.vout)
-    }
-
-    /// Creates new seal using OpretFirst closing method for the provided
-    /// outpoint. Uses `thread_rng` to initialize blinding factor.
-    pub fn opret_first_rand_from(outpoint: Outpoint) -> Self {
-        BlindSeal::opret_first_rand(outpoint.txid, outpoint.vout)
-    }
-
-    /// Creates new seal using TapretFirst closing method for the provided
-    /// outpoint. Uses `thread_rng` to initialize blinding factor.
-    pub fn tapret_first_rand(txid: impl Into<Id>, vout: impl Into<Vout>) -> Self {
-        BlindSeal::with_rng(CloseMethod::TapretFirst, txid, vout, &mut thread_rng())
-    }
+impl<Id: SealTxid> BlindSeal<Id> {
+    /// Creates new seal for the provided outpoint. Uses `thread_rng` to initialize blinding
+    /// factor.
+    pub fn rand_from(outpoint: Outpoint) -> Self { BlindSeal::rand(outpoint.txid, outpoint.vout) }
 
-    /// Creates new seal using OpretFirst closing method for the provided
-    /// outpoint. Uses `thread_rng` to initialize blinding factor.
-    pub fn opret_first_rand(txid: impl Into<Id>, vout: impl Into<Vout>) -> Self {
-        BlindSeal::with_rng(CloseMethod::OpretFirst, txid, vout, &mut thread_rng())
+    /// Creates new seal for the provided outpoint. Uses `thread_rng` to initialize blinding
+    /// factor.
+    pub fn rand(txid: impl Into<Id>, vout: impl Into<Vout>) -> Self {
```

Comment:
This seems to be the same as we have just below in line 141 `new_random`

Reply:
Good catch. I'll keep `new_random` and drop `rand` since the former is used in several places

Reply:
done